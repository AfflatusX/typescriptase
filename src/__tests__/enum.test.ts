/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/tests.ts::enumTest>>
 * BESPOKE<<imports, beforeAll, afterAll, beforeEach, afterEach, enum1, enum2>>
 * SIGNED<<qrWx85WPSS7R1yFaumH3qRdr8OOA0GdSDtTMxIAbbRpPHdVK42Sb/ghnAb1sm1inUdhuUQXjyvESCBrPkxA9mQ==>>
 */

/* BESPOKE START <<imports>> */
import { enum1, enum2 } from "./examples";
/* BESPOKE END <<imports>> */

beforeAll(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeAll>> */
    /* BESPOKE END <<beforeAll>> */
  },
);

afterAll(
  async (): Promise<void> => {
    /* BESPOKE START <<afterAll>> */
    /* BESPOKE END <<afterAll>> */
  },
);

beforeEach(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeEach>> */
    /* BESPOKE END <<beforeEach>> */
  },
);

afterEach(
  async (): Promise<void> => {
    /* BESPOKE START <<afterEach>> */
    /* BESPOKE END <<afterEach>> */
  },
);

test(
  "enum1",
  async (): Promise<void> => {
    /* BESPOKE START <<enum1>> */
    expect(enum1.print({
      name: "enum1",
      path: "src/__tests__/enum.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<enum1>> */
  },
);

test(
  "enum2",
  async (): Promise<void> => {
    /* BESPOKE START <<enum2>> */
    expect(enum2.print({
      name: "enum2",
      path: "src/__tests__/enum.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<enum2>> */
  },
);
