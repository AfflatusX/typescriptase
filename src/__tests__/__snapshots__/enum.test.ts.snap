// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum1 1`] = `
"enum enum1 {
  A = \\"B\\",
  C = \\"D\\",
}
"
`;

exports[`enum2 1`] = `
"export enum enum2 {
  E = 0,
  F = 1,
}
"
`;
